### PR TITLE
docs: add faq regarding the bazel output tree being readonly

### LINKED ISSUE #279: Issue with `readonly` package directory under node_modules
I have a `js_binary` rule that I'm running outside of bazel, where a dependency copies a file from one place to another. However, the directory where the copy happens is `readonly`:

```
  errno: -13,
  syscall: 'copyfile',
  code: 'EACCES',
  path: '/private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modules/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/aws-cdk-lib/core/lib/custom-resource-provider/nodejs-entrypoint.js',
  dest: '/private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modules/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/aws-cdk-lib/aws-s3/lib/auto-delete-objects-handler/__entrypoint__.js'
```

Specifically, the `bazel-out/darwin-fastbuild/bin/node_modules/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/aws-cdk-lib` directory is `readonly`: 

```
╭────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─────────┬──────────╮
│ #  │                                                              name                                                              │  type   │ readonly │
├────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼─────────┼──────────┤
│  0 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ dir     │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/@balena                                                 │         │          │
│  1 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ dir     │ true     │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/aws-cdk-lib                                             │         │          │
│  2 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/case                                                    │         │          │
│  3 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/constructs                                              │         │          │
│  4 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/fs-extra                                                │         │          │
│  5 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/ignore                                                  │         │          │
│  6 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/jsonschema                                              │         │          │
│  7 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/minimatch                                               │         │          │
│  8 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/punycode                                                │         │          │
│  9 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/semver                                                  │         │          │
│ 10 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/node_modul │ symlink │ false    │
│    │ es/.aspect_rules_js/aws-cdk-lib@2.31.1_constructs@10.1.43/node_modules/yaml                                                    │         │          │
╰────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────┴──────────╯
```

Interestingly, another package, `@balena`, is not readonly. Also, going to the main `node_modules` folder, we can see the directory isn't `readonly`:

```
❯ : ls -l /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/deploy/node_modules/ | select name type readonly | where name =~ aws-cdk-lib
╭───┬─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─────────┬──────────╮
│ # │                                                              name                                                               │  type   │ readonly │
├───┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┼─────────┼──────────┤
│ 0 │ /private/var/tmp/_bazel_duarte.nunes/f66c51e6a3394c97f40c1dcafe76e7fd/execroot/umani/bazel-out/darwin-fastbuild/bin/deploy/node │ symlink │ false    │
│   │ _modules/aws-cdk-lib                                                                                                            │         │          │
╰───┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────┴──────────╯
```

Is there any reason for the directory to be `readonly` under `aws-cdk-lib@2.31.1_constructs@10.1.43`?

### PR DESCRIPTION
**Pull request by jbedard**
Fix #279

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -33,6 +33,12 @@ include the `pnpm-workspace.yaml` if it exists and all `package.json` files in y
 To list all local `package.json` files that pnpm needs to read, you can run
 `pnpm recursive ls --depth -1 --porcelain`.
 
+## An npm package is failing when writing to the `node_modules` directory
```
Comment (gregmagolan) [inline docs/faq.md:36]: The issue that this answer originated from was a js_binary tool that was trying to write to a node_modules directory and not an npm package. https://github.com/aspect-build/rules_js/issues/279

I think the question is more general:

```
## Can a tool run outside of Bazel write to the `node_modules` in `bazel-out`?

Some tools such as the AWS SDK write to node_modules when they are run. Ideally this should be avoided or fixed upstream. ...
```
```

Comment (jbedard) [reply docs/faq.md]: Updated

File: js/private/test/image/structure/digests.sum [hunk 2]
```diff
@@ -1,2 +1,2 @@
-099acc2677cf8fbcd5c49d79169a21a8f541234d312e8e71b260af484093e061
+db4b47b2410911b723bed458f4be3ac6eae5859c232e924f7401856224233196
```
Comment (gregmagolan) [inline js/private/test/image/structure/digests.sum:1]: This one again.. it also snuck into one of Derek's PRs.

Comment (gregmagolan) [reply js/private/test/image/structure/digests.sum:1]: Can you undo the change to this file before landing?

**Review by gregmagolan — APPROVED** [review]

File: js/private/test/image/structure/digests.sum (hunk 2 above)
Comment (jbedard) [reply js/private/test/image/structure/digests.sum:1]: The build was red without this. What's the root cause of it?

I'm out atm but will try removing it in a bit

Comment (gregmagolan) [reply js/private/test/image/structure/digests.sum:1]: I'm not sure. It is related to a change @thesayyn made recently. We'll have to sort it out as it is affecting our PRs.

Comment (gregmagolan) [reply js/private/test/image/structure/digests.sum:1]: Looks like Derek fixed it in https://github.com/aspect-build/rules_js/pull/1012 so you may just be able to rebase.

**Pull request merged**